sql/session.go

```diff
@@ -139,6 +139,8 @@ type Session interface {
 	GetCharacterSetResults() CharacterSetID
 	// GetCollation returns the collation for this session (defined by the system variable `collation_connection`).
 	GetCollation() CollationID
+	// ValidateSession validates a working set if there are a valid sessionState with non-nil working set.
```

Comment:
(minor) I don't think we ever talk about a "working set" in the GMS layer, so I'd recommend keeping the documentation a bit more generic at the GMS layer. Something like... 
```suggestion
	// ValidateSession provides integrators a chance to do any custom validation of this session before any query is executed in it. For example, Dolt uses this hook to validate that the session's working set is valid. 
```